test/blas_test.hpp

```diff
@@ -223,6 +223,31 @@ static inline void fill_trsm_matrix(std::vector<scalar_t> &A, size_t k,
   }
 }
 
+/**
+ * @brief Set to zero the last n bits of a float.
+ *
+ * @param val input/output float value.
+ * @param nbits number of last bit set to zero. It is set by default to 13 since
+ * this is the difference of the number of bits of the mantissa between floats
+ * (23) and FP16 / NVIDIA TF32 (10). For bfloat16, this value needs to be set to
+ * 16 to get correct result.
+ */
+static inline void set_to_zero_last_nbits(float &val, int32_t nbits = 13) {
+  int32_t *int_pntr = reinterpret_cast<int32_t *>(&val);
```

Comment:
```suggestion
template <typename T>
void set_to_zero_last_nbits(T &val, int32_t nbits = 13) {
  static_assert(sizeof(T) <= 64);
  using integer_t =
      std::conditional_t<sizeof(T) == 64, int64_t,
                         std::conditional_t<sizeof(T) == 32, int32_t, int16_t>>;
  integer_t *int_pntr = reinterpret_cast<integer_t *>(&val);
```

We need to make this function generic  for compiling in case of `double`s even if we don't use them.

*Same below.*